gluten-core/src/main/scala/io/glutenproject/execution/CartesianProductExecTransformer.scala

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.glutenproject.execution
+
+import io.glutenproject.backendsapi.BackendsApiManager
+import io.glutenproject.extension.ValidationResult
+import io.glutenproject.metrics.MetricsUpdater
+import io.glutenproject.substrait.SubstraitContext
+import io.glutenproject.substrait.rel.RelBuilder
+
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.expressions.{Attribute, Expression}
+import org.apache.spark.sql.catalyst.plans.{Inner, JoinType}
+import org.apache.spark.sql.execution.SparkPlan
+import org.apache.spark.sql.execution.joins.BaseJoinExec
+import org.apache.spark.sql.execution.metric.SQLMetric
+import org.apache.spark.sql.vectorized.ColumnarBatch
+
+// Only supports basic cartesian product with no conditions for now
+case class CartesianProductExecTransformer(
+    left: SparkPlan,
+    right: SparkPlan,
+    condition: Option[Expression])
+  extends BaseJoinExec
+  with TransformSupport {
+
+  override def joinType: JoinType = Inner
+
+  override def leftKeys: Seq[Expression] = Nil
+
+  override def rightKeys: Seq[Expression] = Nil
+
+  override def columnarInputRDDs: Seq[RDD[ColumnarBatch]] =
+    getColumnarInputRDDs(left) ++ getColumnarInputRDDs(right)
+
+  // Note: "metrics" is made transient to avoid sending driver-side metrics to tasks.
+  @transient override lazy val metrics: Map[String, SQLMetric] =
+    BackendsApiManager.getMetricsApiInstance.genCartesianProductTransformerMetrics(sparkContext)
+
+  override def metricsUpdater(): MetricsUpdater = {
+    BackendsApiManager.getMetricsApiInstance.genCartesianProductTransformerMetricsUpdater(metrics)
+  }
+
+  override def doTransform(context: SubstraitContext): TransformContext = {
```

Comment:
need to add a check for the backend, and apply this transform logic only for Velox backed

Reply:
In `CHSparkPlanExecApi`, `UnsupportedOperationException` is being thrown for CartesianProductExec. It will not come into this path in Clickhouse. Is the backend specific check still required?

Reply:
Added backend specific check in "doValidateInternal" and in TransformHintRule.